src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -22,13 +23,24 @@ public class InMemoryExporter<T> : BaseExporter<T>
         where T : class
     {
         private readonly ICollection<T> exportedItems;
+        private readonly ExportDelegate onExport;
```

Comment:
Use `Func<Batch<T>, ExportResult>` instead of a `delegate`? I think that should simplify the assignment operation in the ctor.

Reply:
I can do this change and remove the `Delegate`. 
Note that this will complicate the other ctor as now the `DefaultExport()` method needs to be encapsulated in a func.
`this.onExport = (Batch<T> batch) => this.DefaultExport(batch);`